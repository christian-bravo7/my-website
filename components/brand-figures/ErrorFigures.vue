<template>
  <div
    style="z-index: -1;"
    class="absolute inset-0 | flex justify-center items-center | overflow-hidden pointer-events-none"
  >
    <figure
      v-for="index in 10"
      :key="index"
      :class="`absolute | opacity-50 | primary-text figure--${index}`"
    >
      <QuestionIcon class="w-full h-full | fill-current" />
    </figure>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'nuxt-property-decorator';

import QuestionIcon from '~/assets/img/question-icon.svg?inline';

@Component({
  components: {
    QuestionIcon,
  },
})
export default class ErrorFigures extends Vue {}
</script>
