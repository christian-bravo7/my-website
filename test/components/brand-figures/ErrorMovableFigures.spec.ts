import ErrorFigures from '@/components/brand-figures/ErrorFigures.vue';
import { mount } from '@vue/test-utils';

describe('ErrorFigures component', () => {
  it('should render correclty', () => {
    const wrapper = mount(ErrorFigures);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display 10 question icons', () => {
    const wrapper = mount(ErrorFigures);

    const questionIcons = wrapper.findAll('figure');

    expect(questionIcons.length).toBe(10);
  });
});
