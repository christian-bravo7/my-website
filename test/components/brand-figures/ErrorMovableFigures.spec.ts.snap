// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorFigures component should render correclty 1`] = `
<div class="absolute inset-0 overflow-hidden pointer-events-none flex justify-center items-center">
  <figure class="primary-text opacity-50 absolute figure--1"><svg class="w-full h-full fill-current"></svg></figure>
  <figure class="primary-text opacity-50 absolute figure--2"><svg class="w-full h-full fill-current"></svg></figure>
  <figure class="primary-text opacity-50 absolute figure--3"><svg class="w-full h-full fill-current"></svg></figure>
  <figure class="primary-text opacity-50 absolute figure--4"><svg class="w-full h-full fill-current"></svg></figure>
  <figure class="primary-text opacity-50 absolute figure--5"><svg class="w-full h-full fill-current"></svg></figure>
  <figure class="primary-text opacity-50 absolute figure--6"><svg class="w-full h-full fill-current"></svg></figure>
  <figure class="primary-text opacity-50 absolute figure--7"><svg class="w-full h-full fill-current"></svg></figure>
  <figure class="primary-text opacity-50 absolute figure--8"><svg class="w-full h-full fill-current"></svg></figure>
  <figure class="primary-text opacity-50 absolute figure--9"><svg class="w-full h-full fill-current"></svg></figure>
  <figure class="primary-text opacity-50 absolute figure--10"><svg class="w-full h-full fill-current"></svg></figure>
</div>
`;
